src/BulkyItem/FileItem.php

```diff
@@ -53,12 +53,24 @@ public static function restore($contents, array $meta): BulkyItemInterface
         return new self($contents, $meta['name'], $meta['type'], $meta['size']);
     }
 
-    public static function fromPath(string $path, string $name, string $mimeType, int $size): self
+    public static function fromPath(string $path, string|null $name = null, string|null $mimeType = null, int|null $size = null): self
     {
         if (!(new Filesystem())->exists($path)) {
             throw new \InvalidArgumentException(\sprintf('The file "%s" does not exist.', $path));
         }
 
+        if (null === $name) {
+            $name = basename($path);
+        }
+
+        if (null === $mimeType) {
+            $mimeType = mime_content_type($path);
```

Comment:
May be use the mime type guesser from Symfony here instead?

Reply:
any hint what that looks like?

Reply:
You could instantiate it yourself via `(new MimeTypes())->guessMimeType($path)`. However, the `symfony/framework-bundle` also registers it as the `mime_types` service, which allows the system to add additional guessers via the `MimeTypeGuesserInterface` via the `AddMimeTypeGuesserPass` compiler pass. This would have to be done via a helper service outside this static function.